src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/KeyReadsByVariantShard.java

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.VariantShard;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.List;
+
+/**
+ * KeyReadByVariantShard takes a PCollection of reads and keys each read by the variants shards it overlaps. Note that
+ * if a read span several shards multiple KVs will be produced. This part of the larger transform to join reads with
+ * overlapping variants.
+ *
+ * |---- shard 0 -----|---- shard 1 -----|---- shard 2 -----|---- shard 3 -----|---- shard 4 -----|
+ *           |-------------- read 1 --------------|
+ *  results in
+ *  KV<shard 0, read 1>
+ *  KV<shard 1, read 1>
+ *  KV<shard 2, read 1>
```

Comment:
Added an additional comment.  There was a copy-paste problem in the previous one.